pkg/reconciler/route/resources/ingress.go

```diff
@@ -134,33 +138,17 @@ func getChallengeHosts(challenges []v1alpha1.HTTP01Challenge) map[string]v1alpha
 	return c
 }
 
-func routeDomains(ctx context.Context, targetName string, r *servingv1alpha1.Route, visibility netv1alpha1.IngressVisibility) ([]string, error) {
+func routeDomain(ctx context.Context, targetName string, r *servingv1alpha1.Route, visibility netv1alpha1.IngressVisibility) (string, error) {
 	hostname, err := domains.HostnameFromTemplate(ctx, r.Name, targetName)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
 	meta := r.ObjectMeta.DeepCopy()
-	labels.SetVisibility(meta, true)
-	clusterLocalName, err := domains.DomainNameFromTemplate(ctx, *meta, hostname)
-	if err != nil {
-		return nil, err
-	}
-	ruleDomains := []string{clusterLocalName}
-
-	if visibility != netv1alpha1.IngressVisibilityClusterLocal {
-		labels.SetVisibility(meta, false)
-		fullName, err := domains.DomainNameFromTemplate(ctx, *meta, hostname)
-		if err != nil {
-			return nil, err
-		}
-
-		if fullName != clusterLocalName {
```

Comment:
Looks like in your code, this case is omitted. Could `fullName != clusterLocalName` happen?

Reply:
The new method only returns either fullName or clusterLocalName.  But in the outer loop we choose to call this once or twice.